PIL/SpiderImagePlugin.py

```diff
@@ -174,8 +174,8 @@ def tell(self):
     def seek(self, frame):
         if self.istack == 0:
             raise EOFError("attempt to seek in a non-stack file")
-        if frame >= self._nimages:
-            raise EOFError("attempt to seek past end of file")
+        if not self._seek_check(frame):
+            return
```

Comment:
I'm going to presume that you meant 'self.n_frames', rather than 'self._n_frames', making this the same comment as you made on the MpoImagePlugin - if this is not the case, let me know. If it is the case, then the n_frames property method returns self._nimages, making them equivalent.